common/string_helpers_test.cpp

```diff
@@ -30,6 +30,7 @@ TEST(UnescapeStringLiteral, Valid) {
   EXPECT_THAT(UnescapeStringLiteral("\\x12"), Optional(Eq("\x12")));
   EXPECT_THAT(UnescapeStringLiteral("test", 1), Optional(Eq("test")));
   EXPECT_THAT(UnescapeStringLiteral("test\\#n", 1), Optional(Eq("test\n")));
+  EXPECT_THAT(UnescapeStringLiteral("r\\u{E9}al\\u{2764}\\u{FE0F}\\u{1F50A}!"), Optional(Eq("réal❤️🔊!")));
```

Comment:
After looking between this and the design, I've filed https://github.com/carbon-language/carbon-lang/pull/2040 to propose a restriction on the number of digits.

Because that proposal isn't yet accepted, I'd suggest adding tests for both `\u{}` and `\u{00000000E9}`. However, whether those are valid (matching the current design) or invalid (matching #2040) I think is up to you -- regardless, the behavior should be tested and consistent.

Reply:
Added test cases for the above-mentioned.

Reply:
It looks like #2040 is being accepted, so please make sure those cases are correctly handled as invalid.

Reply:
The tests in common/string_helpers_test.cpp, in the block TEST(UnescapeStringLiteral, InvalidUnicodes) should address these as unit tests.